test/plugin_functional/plugins/core_plugin_b/public/plugin.tsx

```diff
@@ -34,6 +35,15 @@ export class CorePluginBPlugin
   implements Plugin<CorePluginBPluginSetup, CorePluginBPluginStart, CorePluginBDeps> {
   public setup(core: CoreSetup, deps: CorePluginBDeps) {
     window.corePluginB = `Plugin A said: ${deps.core_plugin_a.getGreeting()}`;
+
+    core.application.register({
+      id: 'bar',
+      title: 'Bar',
+      async mount(context, params) {
```

Comment:
we need to create recommendation how to declare additional properties on `context`. if we delete this import https://github.com/elastic/kibana/pull/41007/files#diff-73111fdf9de323e24d3cf6695ad53aafR21 `context` doesn't contain methods provided by dependency. because dependency declaration is not imported 
```ts
declare module 'kibana/public' {
  interface AppMountContext {...}
}
```